ui/apps/platform/src/services/ComplianceScanConfigurationService.ts

```diff
@@ -0,0 +1,173 @@
+    description?: string;
+};
+
+export type ComplianceScanConfiguration = {
+    id?: string;
+    scanName: string;
+    scanConfig: BaseComplianceScanConfigurationSettings;
+    clusters: string[];
+};
+
+export type ComplianceScanConfigurationStatus = {
+    id: string;
+    scanName: string;
+    scanConfig: BaseComplianceScanConfigurationSettings;
+    clusterStatus: ClusterScanStatus[];
+    createdTime: string; // ISO 8601 date string;
+    lastUpdatedTime: string; // ISO 8601 date string;
+    modifiedBy: SlimUser;
+};
+
+/*
+ * Fetches a list of scan configurations.
+ */
+export function listComplianceScanConfigurations(
+    sortOption: ApiSortOption,
+    page?: number,
+    pageSize?: number
+): Promise<ComplianceScanConfigurationStatus[]> {
+    let offset: number | undefined;
+    if (typeof page === 'number' && typeof pageSize === 'number') {
+        offset = page > 0 ? page * pageSize : 0;
+    }
+    const query = {
+        pagination: { offset, limit: pageSize, sortOption },
+    };
+    const params = qs.stringify(query, { arrayFormat: 'repeat', allowDots: true });
+    return axios
+        .get<{
+            configurations: ComplianceScanConfigurationStatus[];
+        }>(`${complianceScanConfigBaseUrl}?${params}`)
+        .then((response) => {
+            return response?.data?.configurations ?? [];
+        });
+}
+
+/*
+ * Fetches a scan configuration by ID.
+ */
+export function getComplianceScanConfiguration(
+    scanConfigId: string
+): CancellableRequest<ComplianceScanConfigurationStatus> {
+    return makeCancellableAxiosRequest((signal) =>
+        axios
+            .get<ComplianceScanConfigurationStatus>(
+                `${complianceScanConfigBaseUrl}/${scanConfigId}`,
+                {
+                    signal,
+                }
+            )
+            .then((response) => response.data)
+    );
+}
+
+/*
+ * Creates a new scan configuration or updates an existing one.
+ */
+export function saveScanConfig(
+    complianceScanConfiguration: ComplianceScanConfiguration
+): Promise<ComplianceScanConfiguration> {
+    const promise = complianceScanConfiguration.id
+        ? axios.put<ComplianceScanConfiguration>(
+              `${complianceScanConfigBaseUrl}/${complianceScanConfiguration.id}`,
+              complianceScanConfiguration
+          )
+        : axios.post<ComplianceScanConfiguration>(
+              complianceScanConfigBaseUrl,
+              complianceScanConfiguration
+          );
+
+    return promise.then((response) => response.data);
+}
+
+/*
+ * Deletes a scan configuration by ID.
+ */
+export function deleteComplianceScanConfiguration(scanConfigId: string) {
+    return axios
+        .delete<Empty>(`${complianceScanConfigBaseUrl}/${scanConfigId}`)
+        .then((response) => {
+            return response.data;
+        });
+}
+
+/*
+ * Returns the count of scan configurations.
+ */
+export function getComplianceScanConfigurationsCount(): Promise<number> {
```

Comment:
For your info for possible future refactoring, response of `listComplianceScanConfigurations` request now has `totalCount` property.